docs/bert_for_longer_texts.md

```diff
@@ -1,24 +1,34 @@
 # BERT modification for longer texts
 
 ## Motivation
-BERT model can only use the text of the maximal length of 512 tokens (roughly speaking: token = word). It is built in the model architecture and cannot be directly changed. Discussion of this issue can be found [here](https://github.com/google-research/bert/issues/27).
+BERT model can only use the text of the maximal length of $512$ tokens (roughly speaking: token = word). It is built in the model architecture and cannot be directly changed. Discussion of this issue can be found [here](https://github.com/google-research/bert/issues/27).
 
 ## Method
 Method to overcome this issue was proposed by Devlin (one of the authors of BERT) in the aforementioned discussion: [comment](https://github.com/google-research/bert/issues/27#issuecomment-435265194)
 According to this, we do the following steps
-1. Prepare the single text - we follow [this instruction](https://www.kdnuggets.com/2021/04/apply-transformers-any-length-text.html)
-- tokenize all text
-- divide all tokens into chunks of the size `size`, overlapping with step `step`, with minimal chunk length `minimal_length`
-- for each chunk we add special tokens at the beginning and the end
-- add padding tokens to make all tokenized sequences the same length
-- stack the tensor chunks into one via `torch.stack`
+1. Prepare the single text
+- We follow [this instruction](https://www.kdnuggets.com/2021/04/apply-transformers-any-length-text.html). The key difference is that we allow the text chunks to overlap.
+The splitting procedure is the following:
+- The procedure of splitting and pooling is determined by the hyperparameters of the class `BertClassifierWithPooling`. These are `chunk_size`, `stride`, `minimal_chunk_length`, `maximal_text_length` and `pooling_strategy`.
+- Tokenize the whole text (if `maximal_text_length=None`) or truncate to the size `maximal_text_length`.
+- Split the tokens to chunks of the size `chunk_size`.
+- Tokens may overlap dependent on the parameter `stride`.
+- In other words: we get chunks by moving the window of the size `chunk_size` by the length equal to `stride`.
+- See the example in [the comment](https://github.com/google-research/bert/issues/27#issuecomment-435265194).
+- More examples of splitting with differents sets of parameters are in [test_splitting](../tests/model/test_splitting.py).
+- Stride has the analogous meaning here that in convolutional neural networks.
+- The `chunk_size` is analogous to kernel_size in CNNs.
+- We ignore chunks which are too small - smaller than `minimal_chunk_length`.
+- For each chunk add special tokens at the beginning and the end.
+- Add padding tokens to make all tokenized sequences the same length.
+- Stack the tensor chunks into one via `torch.stack`.
```

Comment:
I think it would be easier to read if we separate general description, description of parameters and steps. In particular, I don't like that the list continues in the same way after `The splitting procedure is the following:`.

I propose something like this. We can even move general description and/or description of parameters to `## Method`.
```
1. Prepare the single text
We follow [this instruction](https://www.kdnuggets.com/2021/04/apply-transformers-any-length-text.html). The key difference is that we allow the text chunks to overlap.

The procedure of splitting and pooling is determined by the hyperparameters of the class `BertClassifierWithPooling`. These are `chunk_size`, `stride`, `minimal_chunk_length`, `maximal_text_length` and `pooling_strategy`.
<description of each parameter>

The splitting procedure is the following:
- step 1
- step 2
- ...
```

We can also mark each point with a smaller heading (`###`).